src/toil/batchSystems/abstractBatchSystem.py

```diff
@@ -139,6 +146,22 @@ def _getResultsFileName(self, toilPath):
         """
         return os.path.join(toilPath, "results.txt")
 
+    @staticmethod
+    def supportsWorkerCleanup():
+        """
+        Indicates whether the batch system supports the cleanup of workers on shutdown
+        """
+        raise NotImplementedError('Abstract method: supportsWorkerCleanup')
```

Comment:
Can you remove the message argument to the exception constructor? It really is redundant and bound to become stale (I've seen this happen again and again). The stack trace will make it clear which method was not implemented.